Libraries/BatchedBridge/MessageQueue.js

```diff
@@ -150,22 +150,15 @@ class MessageQueue {
   enqueueNativeCall(moduleID: number, methodID: number, params: Array<any>, onFail: ?Function, onSucc: ?Function) {
     if (onFail || onSucc) {
       if (__DEV__) {
-        const callId = this._callbackID >> 1;
-        this._debugInfo[callId] = [moduleID, methodID];
-        if (callId > DEBUG_INFO_LIMIT) {
-          delete this._debugInfo[callId - DEBUG_INFO_LIMIT];
+        this._debugInfo[this._callID] = [moduleID, methodID];
+        if (this._callID > DEBUG_INFO_LIMIT) {
+          delete this._debugInfo[this._callID - DEBUG_INFO_LIMIT];
         }
       }
-      onFail && params.push(this._callbackID);
-      /* $FlowFixMe(>=0.38.0 site=react_native_fb,react_native_oss) - Flow error
-       * detected during the deployment of v0.38.0. To see the error, remove
-       * this comment and run flow */
-      this._callbacks[this._callbackID++] = onFail;
-      onSucc && params.push(this._callbackID);
-      /* $FlowFixMe(>=0.38.0 site=react_native_fb,react_native_oss) - Flow error
-       * detected during the deployment of v0.38.0. To see the error, remove
-       * this comment and run flow */
-      this._callbacks[this._callbackID++] = onSucc;
+      onFail && params.push(this._callID * 2);
```

Comment:
Rather than mixing math (multiplying, modulo) and bitwise ops can you use just bitwise ops when encoding & decoding the call IDs, I think it will be clearer in the code what is going on.